ARM64:dts:aspeed Congo DTS changes for A1

aasboddu opened the pull request:
Update DTS entries according to change in A1 Aspeed BMC chip. Following are the changes.
- BIOS/FPGA flash change to SPI2 controller
- I2C Bus 10  change to I2C Bus 12
- USB VHUB change to XHCI
- eSPI GPIO change for POST Codes

tested: verified in Congo platform

aasboddu committed:
ARM64:dts:aspeed:Congo: Add BIOS/FPGA flash memory

Add SPI2 controller for BIOS/FPGA flash memory
access. Updated clock speed to 33MHz

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Congo: Modify I2C Bus for Congo A1

Modify I2C Bus number for Congo based on
A1 silicon specification.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-g7.dtsi:
```diff
@@ -1199,15 +1199,32 @@
-			num-cs = <2>;
-			status = "disabled";
-		};
+                spi2: spi@14030000 {
```
This is apseed owned dtsi file. any custom change here can create confuison future.
Cleaner approch is to move the amd specific chnages in amd machine specific dts or work with aspeed , incase you need to keep this in a1 specific dtsi.

aasboddu replied:
moved changes to dts

aasboddu committed:
ARM64:dts:aspeed:Congo: Add USB vHUB for KVM

Add USB vHUB to enable keyboard/mouse for KVM

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

ojayanth commented:
Please include summary information in the PR description.

rajaganeshr commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -799,10 +799,6 @@ spd_ ## bus ## _ ## index: spd@addr,4cc5118 ## index ## 000 { \
         status = "okay";
 };
 
-&ltpi0 {
```
is this not required for LTPI? is it added in a different file?

aasboddu replied:
Duplicate entry

aasboddu commented:
> Please include summary information in the PR description.

I have added the summary

ojayanth commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -226,12 +226,26 @@
+                reg = <1>;
+                status = "okay";
+                m25p,fast-read;
+                label = "pnor";
```
Lable name should be diffrent?

srgovard commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -804,6 +800,19 @@ spd_ ## bus ## _ ## index: spd@addr,4cc5118 ## index ## 000 { \
+};
+
+&usb3ahp {
+       status = "okay";
```
As per ast2700-evb.dts we have pinctrl-0 defines for 'usb3ahp', why they are missing here?
&usb3ahp {
	status = "okay";
	**pinctrl-0 = <&pinctrl_usb3axhp_default &pinctrl_usb2axhp_default>**;
};

aasboddu committed:
ARM64:dts:aspeed:Congo: Add GPIO HOG for eSPI

Add GPIO HOG to set eSPI GPIO line to low and
direction to output enabling BMC to receive BIOS
POST codes for A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

srgovard reviewed (commented):
Please take care of spaces and replace with tab.

srgovard commented on arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-congo.dts:
```diff
@@ -41,38 +41,38 @@
 		};
 
 		espi_mmbi_memory: espi-mmbi-memory@428000000 {
```
ast2700-evb has defined espi0_mmbi_memory and espi1_mmbi_memory. The offset here belongs to espi1. This may not be needed now, but might needed when we enable espi0 or 1. Check with Ashok/Raja.

aasboddu replied:
Discussed with Raja and ashok these are not needed for now.

ashok2ku committed:
Update aspeed-bmc-amd-congo.dts

As per Raja/Sreedhar - we dont need this changes for now
